can/message.py

```diff
@@ -130,7 +137,10 @@ def __init__(self, timestamp=0.0, arbitration_id=0, is_extended_id=None,
             self.dlc = dlc
 
         if check:
-            self._check()
+            try:
+                self._check()
+            except AssertionError as error:
+                raise ValueError(str(error))
```

Comment:
One should not rely on assertions as they may be ignored in production code. Maybe check() should raise ValueError directly.

Reply:
Ah yeah, I thought about that as well ... I'll do that before we merge this. 👍